README.md

```diff
@@ -66,7 +66,47 @@ dotnet format <path_to_your_solution_or_project> --severity warn --verbosity dia
 
 You can also modify the command to specify the IDs of the analysis rules you wish to automatically correct (if a fix is available). In this way, you avoid manual work, and breaking the correction into several pull requests will increase developers' confidence in the process of adopting new standards.
 
-All rules included in this package **can be disabled or modified** in an `.editorconfig` file that you can add to your project. You can also **disable a rule for a specific line or block of code** using `#pragma` directives or `[SuppressMessage]` attributes. Learn more about [configuring code analysis rules](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/suppress-warnings). Remember to always justify why a rule is disabled or modified.
+All rules included in this package **can be disabled or modified** in an `.editorconfig` file that you can add to your project. You can also **disable a rule for a specific line or block of code** using `#pragma` directives or `[SuppressMessage]` attributes. Learn more about [configuring code analysis rules](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/suppress-warnings). Remember to always justify why a rule is disabled or modified. Here're some examples:
+
+- Disable specific diagnostics in the code
+
+    ````c#
+    #pragma warning disable CA2200 // Rethrow to preserve stack details
+    throw e;
+    #pragma warning restore CA2200
+    ````
+
+- Disable specific diagnostics for a method
+
+    ````c#
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "CA2200:Rethrow to preserve stack details", Justification = "Not production code.")]
+    void Sample()
+    {
+    }
+    ````
+
+- Disable a rule using an .editorconfig
+
+    ````editorconfig
+    # Disable CA2200 globally
+    [*.cs]
+    dotnet_diagnostic.CA2200.severity = none
+
+    # Disable CA2200 for specific files
+    [tests/**/*.cs]
+    dotnet_diagnostic.CA2200.severity = none
+    ````
+
+Finally, the warnings breaks the CI because they are treated as errors when building the `Release` configuration. You can disable this behavior by setting the following MSBuild property in the `.csproj` file or in the `Directory.Build.props` file
```

Comment:
Should we really mention this? Feels like people will just do this and then magic no more errors so they do not look at their warning. 

Of maybe mention this shouldn't be a permanent solution?

Reply:
just my 2 cent as a humble user, I think it's fine in specific use-case, considering that we are using more and more code generator, that makes sens to not crash a build because the generated client is not on par with our coding standards.

as such having the doc tell us about how to manage that situation is useful. maybe add a few example expected use-case to help steer the dev decision a bit?

Reply:
> using more and more code generator, that makes sens to not crash a build because the generated client is not on par with our coding standards.

The WarningsAsErrors property is global to a project, so we cannot recommend disable it on the long term. It should only be a temporary solution.

If a source generator is not compliant, then you should update the generator.
Also, Roslyn has the concept of generated files. These files are excluded from analysis (except a specific configuration from an analyzer). There are many ways to indicate a file is generated (header, editorconfig, MSBuild property).
Finally, you can disable rules per file/folder using an editorconfig. So, you could disable some rules on the generated files.